.circleci/check_gotest.sh

```diff
@@ -2,14 +2,29 @@
 
 # Run test in each go environment and log errors to $FILEPATH_ERROR_FLAG.
 # If all tests pass successfully then the file will not be created at all.
-export GOPATH=$HOME/$GOPATH_FOLDER;
 # cd to code with go.mod file outside of GOPATH to work with module-based behaviour
 # https://github.com/golang/go/wiki/Modules#when-do-i-get-old-behavior-vs-new-module-based-behavior
 cd $HOME/project
```

Comment:
BTW, don't we already start in this directory? I thought that CircleCI launches the scripts from the source tree directory.

Reply:
Just checked, CircleCI does `cd` into that dir. And it is also specified in env `CIRCLE_WORKING_DIRECTORY=~/project` (that's a default value, see [docs](https://circleci.com/docs/2.0/configuration-reference/#job_name)).
So should we just remove `cd $HOME/project` or put `cd $(eval echo $CIRCLE_WORKING_DIRECTORY)` instead to be sure? We would need that `eval` since it contains `~` which needs to be resolved. I guess just removing `cd` is simpler and won't break anything.

Reply:
> So should we just remove cd $HOME/project

I think this will be the most correct approach. If we don't change the working directory, we can expect it to be the directory with checked out source tree, hence this `cd` is redundant. Removing it will make the script more readable, IMO.

> since it contains ~ which needs to be resolved.

(My Bash works just fine with tildes in variables :thinking:)

Reply:
Oh... Yeah, you're right! Stupid me. I tested only `DIR=~` :facepalm: